### ISSUE TITLE
Translations interface error when no permission to some languages

### ISSUE THREAD

**Issue by LevinGermann** [directus/directus#13009]
### Describe the Bug

We use translations quite a lot in Directus.
Since we have a couple of languages set up in the system (e.g. English, German, French) and only a certain subset is needed at a given time, users can set their "Available Languages" in their user profile:

![image](https://user-images.githubusercontent.com/9347206/165298155-38822769-94f6-40a3-a56f-bb1f88580e12.png)

languages_available is a simple table containing the list of the languages we have set up:

![image](https://user-images.githubusercontent.com/9347206/165298350-f5d8c66b-08da-4a43-a879-a0b87ff819b3.png)

If a user is working on an object in Directus and only wants to see the english translations, we made this work by restricting the READ permission on the Language table with this rule:

![image](https://user-images.githubusercontent.com/9347206/165298945-fb41137c-1452-44e6-8e6b-0c28b2420a88.png)
`Code is one of $CURRENT_USER.languages_available.languages_available_id.code`

Before Directus 9.9.1 this was working fine and since users had all read permissions for languages_available, they could individually set their wanted languages that are being displayed in all translation widgets.

Now this is unfortunately no longer working:
Let's say a user set their languages available to English and Portuguese and want to display an object which contains english and german translations, the translations widget does no longer display anything and the console logs an error:
![image](https://user-images.githubusercontent.com/9347206/165299805-a51202e9-0b43-44d1-b5f3-2ec3dca63104.png)
```
TypeError: Cannot read properties of null (reading 'code')
    at index.dd44fb28.entry.js:518:4242
    at Array.find (<anonymous>)
    at B (index.dd44fb28.entry.js:518:4228)
    at index.dd44fb28.entry.js:518:4974
    at Proxy.map (<anonymous>)
    at Gv.fn (index.dd44fb28.entry.js:518:4876)
    at Gv.run (vendor.c37f7cab.js:2:12019)
    at uH.get value [as value] (vendor.c37f7cab.js:2:20264)
    at Gv.fn (index.dd44fb28.entry.js:518:4505)
    at Gv.run (vendor.c37f7cab.js:2:12019)
```

My first thought was that something with the rule for the languages table might have been broken in an update, so I tried a more straight-forward like to something hardcoded like:
![image](https://user-images.githubusercontent.com/9347206/165300176-e2e229a2-23e9-4c14-b6e6-ca9b84d2e0a7.png)

This led to a Unexpected Error window appear when a user opens the same object as before
```
{
  "stack": "TypeError: Cannot read properties of undefined (reading 'code')\n    at le (https://URL.com/admin/assets/index.dd44fb28.entry.js:518:5773)",
  "message": "Cannot read properties of undefined (reading 'code')"
}
```

### To Reproduce

Steps described in the bug report above.
Should be enough to create a translation and setup a similar environment so that users can edit their wanted languages, then restrict READ access to languages with either
`Code is one of 'de-DE', 'en-US'`
or our original form
`Code is one of $CURRENT_USER.languages_available.languages_available_id.code`

### What version of Directus are you using?

9.9.1

### What version of Node.js are you using?

v16.7.0

### What database are you using?

PostgresQL 13.2

### What browser are you using?

Chrome

### How are you deploying Directus?

Docker

**Comment by LevinGermann** [comment]
Side note, a user has all permissions to the junction table which has been generated for this M2M relationship:

![image](https://user-images.githubusercontent.com/9347206/165303030-74820622-a515-4e17-873b-bd0a837841bb.png)

**Comment by LevinGermann** [comment]
> > Before Directus 9.9.1
> 
> Just to confirm, this also includes `9.9.0` right? Wanted to be sure since there was big relational interfaces rework in that version. Unless of course you meant it breaks after moving _from_ `9.9.0` to `9.9.1` instead of from an even earlier version.

We upgraded directly from 9.4.2.
But I just downgraded to 9.9.0, the issue is there as well.
I then checked on the latest version before 9.9.0, which is 9.8.0. The issue is not there. :) So seems like this issue began to appear in 9.9.0.

On 9.8.0, everything works as expected and I'm getting a translation widget with only my available language, which contains nothing:
![image](https://user-images.githubusercontent.com/9347206/165312810-1c922a82-d6ce-4b1f-bb55-919e96edfdf3.png)
